src/dataflow/src/render/reduce.rs

```diff
@@ -606,3 +631,89 @@ fn is_min_or_max(func: &AggregateFunc) -> bool {
         _ => false,
     }
 }
+
+/// Is the aggregate function is a "max" variant.
+fn is_max(func: &AggregateFunc) -> bool {
+    match func {
+        AggregateFunc::MaxInt32
+        | AggregateFunc::MaxInt64
+        | AggregateFunc::MaxFloat32
+        | AggregateFunc::MaxFloat64
+        | AggregateFunc::MaxDecimal
+        | AggregateFunc::MaxBool
+        | AggregateFunc::MaxString
+        | AggregateFunc::MaxDate
+        | AggregateFunc::MaxTimestamp
+        | AggregateFunc::MaxTimestampTz => true,
+        _ => false,
+    }
+}
+
+/// Monoids for in-place compaction of monotonic streams.
+pub mod monoids {
+    use repr::{Datum, Row};
+    use serde::{Deserialize, Serialize};
+
+    /// A monoid containing a single-datum row, that is updated by SQL's `min`.
+    #[derive(Ord, PartialOrd, Eq, PartialEq, Debug, Clone, Serialize, Deserialize, Hash)]
+    pub struct MinMonoid {
+        pub value: Row,
```

Comment:
just brainstorming: if you wanted to extend this to min-k rather than just min would the right idea be to store a set of rows here + k here and implement something like "join-k" (in the lattice sense) in add-assign?

(its unclear to me how you would recover the values at the end)

Reply:
Yes, that is right. The `count` method should recover the result! The downside is that if you change one input value, you may need to screw around with `k` values. Maybe not worse than normal top k stuff though.